app/src/main/java/com/infomaniak/mail/views/itemViews/BannerWithActionView.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.views.itemViews
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isVisible
+import com.infomaniak.lib.core.utils.getAttributes
+import com.infomaniak.lib.core.utils.setMarginsRelative
+import com.infomaniak.mail.R
+import com.infomaniak.mail.databinding.ViewBannerWithActionBinding
+
+class BannerWithActionView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding by lazy { ViewBannerWithActionBinding.inflate(LayoutInflater.from(context), this, true) }
+
+    var description: CharSequence?
+        get() = binding.description.text
+        set(value) {
+            binding.description.text = value
+        }
```

Comment:
Use delegation if the signatures are the exact same ones

```suggestion
    var description: CharSequence? by binding.actionButton::text
```

Reply:
`var description: CharSequence? by binding.description::text` and not `actionButton`. Will do the modification

Reply:
Yeah true nice catch